Jellyfin.Server.Implementations/Devices/DeviceManager.cs

```diff
@@ -135,7 +135,20 @@ public ClientCapabilities GetCapabilities(string? deviceId)
         /// <inheritdoc />
         public DeviceInfoDto? GetDevice(string id)
         {
-            var device = _devices.Values.Where(d => d.DeviceId == id).OrderByDescending(d => d.DateLastActivity).FirstOrDefault();
+            using var enumerator = _devices
+                .GetEnumerator();
+            Device? device = null;
+            DateTime? deviceLastActivity = null;
+            while (enumerator.MoveNext())
+            {
+                if (enumerator.Current.Value.DeviceId == id && (deviceLastActivity == null || deviceLastActivity < enumerator.Current.Value.DateLastActivity))
+                {
+                    device = enumerator.Current.Value;
+                    deviceLastActivity = enumerator.Current.Value.DateLastActivity;
+                    break;
```

Comment:
Should remove this, since we don't want to break out of the loop on the first match.